api/src/main/java/org/pmiops/workbench/db/dao/NotebookCohortCacheServiceImpl.java

```diff
@@ -0,0 +1,124 @@
+package org.pmiops.workbench.db.dao;
+
+import com.google.common.annotations.VisibleForTesting;
+import org.pmiops.workbench.config.WorkbenchConfig;
+import org.pmiops.workbench.db.model.NotebookCohortCache;
+import org.pmiops.workbench.db.model.User;
+import org.pmiops.workbench.db.model.Workspace;
+import org.pmiops.workbench.db.model.WorkspaceUserRole;
+import org.pmiops.workbench.exceptions.BadRequestException;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import javax.inject.Provider;
+import java.sql.Timestamp;
+
+@Service
+public class NotebookCohortCacheServiceImpl implements NotebookCohortCacheService {
+
+  private Provider<WorkbenchConfig> configProvider;
+
+  @Autowired
+  NotebookCohortCacheDao notebookCohortCacheDao;
+  @Autowired
+  WorkspaceUserRoleDao workspaceUserRoleDao;
+
+  public NotebookCohortCacheServiceImpl(Provider<WorkbenchConfig> configProvider) {
+    this.configProvider = configProvider;
+  }
+
+  @Override
+  public NotebookCohortCacheDao getDao() {
+    return notebookCohortCacheDao;
+  }
+
+  public WorkspaceUserRoleDao getworkspaceUserRoleDao() {
+    return workspaceUserRoleDao;
+  }
+
+  /**
+   * Checks if notebook for given workspace and user is already in cache
+   * if yes, update the lastmodified time only
+   * If no, check the number of cache entries for given workspace and user
+   * if it is above 10, delete the row with least lastAccessTime and add a new entry
+   * @param workspace
+   * @param user
+   * @param notebookName
+   * @param lastAccessDateTime
+   */
+  @Override
+  public void updateNotebook(Workspace workspace, User user, String notebookName,
+      Timestamp lastAccessDateTime) {
+    WorkspaceUserRole workspaceUserRole =
+        getworkspaceUserRoleDao().findWorkspaceUserRolesByWorkspaceAndUser(workspace, user);
+    if (workspaceUserRole == null) {
+      throw new BadRequestException(String.format(
+          "No workspace %s was found for user %s ", workspace.getName(), user.getUserId()));
+    }
+    NotebookCohortCache cache = getDao().findByUserWorkspaceIdAndNotebookName
+        (workspaceUserRole, notebookName);
```

Comment:
I don't have a particular style guide to cite here but I would definitely not wrap the parens on a function call like this. Just moving the first paren up one line would work fine.